activesupport/lib/active_support/deprecation/reporting.rb

```diff
@@ -5,8 +5,6 @@ class Deprecation
     module Reporting
       # Whether to print a message (silent mode)
       attr_accessor :silenced
-      # Name of gem where method is deprecated
-      attr_accessor :gem_name
```

Comment:
I'll just revert this change. I noticed two other accessors defined on other modules at well. I feel they should be part of the main module given that it directly uses them in the initializer, but it's kind of out of scope for this PR.